pydantic_ai_slim/pydantic_ai/ui/web/app.py

```diff
@@ -0,0 +1,175 @@
+"""Factory function for creating a web chat app for a Pydantic AI agent."""
+
+from __future__ import annotations
+
+import os
+import re
+from collections.abc import Mapping, Sequence
+from pathlib import Path
+from typing import TypeVar
+
+import httpx
+from starlette.applications import Starlette
+from starlette.requests import Request
+from starlette.responses import HTMLResponse, Response
+
+from pydantic_ai import Agent
+from pydantic_ai.builtin_tools import AbstractBuiltinTool
+from pydantic_ai.models import KnownModelName, Model, infer_model
+
+from .api import AIModel, add_api_routes
+
+
+def format_model_display_name(model_name: str) -> str:
```

Comment:
Nice :)

Reply:
Some model names have `/` in them, like when you're using OpenRouter or something. If we're goign to do this, test it against all of `KnownModelNames` and see if there are any other rules we should add.

Also, this sounds like should be in `_utils`